Client/Frontend/Browser/BrowserViewController/BrowserViewController+WebViewDelegates.swift

```diff
@@ -269,18 +269,17 @@ extension BrowserViewController: WKNavigationDelegate {
         // always allow this. Additionally, data URIs are also handled just like normal web pages.
 
         if ["http", "https", "data", "blob", "file"].contains(url.scheme) {
-            if navigationAction.navigationType == .linkActivated {
-                resetSpoofedUserAgentIfRequired(webView, newURL: url)
-            } else if navigationAction.navigationType == .backForward {
-                restoreSpoofedUserAgentIfRequired(webView, newRequest: navigationAction.request)
-            }
-            
             if let host = url.host, navigationAction.targetFrame?.isMainFrame ?? false {
                 tab.desktopSite = Tab.DesktopSites.hostList.contains(host)
                 if !tab.desktopSite, tab.isPrivate {
                     // Private mode has an additional memory-only list to check.
                     tab.desktopSite = Tab.DesktopSites.privateModeHostList.contains(host)
                 }
+
+                let requestUA = navigationAction.request.value(forHTTPHeaderField: "User-Agent") ?? ""
+                if UserAgent.isDesktop(ua: requestUA) != tab.desktopSite {
+                    webView.load(URLRequest(url: url, cachePolicy: .reloadIgnoringLocalCacheData, timeoutInterval: 60) as URLRequest)
```

Comment:
this `load` seems to work well with the decisionHandler(.allow), which is suprising to me. The history stack stays correct.